### PR TITLE
Refactor and added new action bigquery-insert-rows

### PR DESCRIPTION
**Pull request by jcortes**
When trying to list datasets I get the following error:
```
Unable to detect a Project Id in the current environment. To learn more about authentication and Google APIs, visit: https://cloud.google.com/docs/authentication/getting-started
```

Reported issues:
* https://pipedream-users.slack.com/archives/CPTJYRY5A/p1653301425468049
* https://pipedream-users.slack.com/archives/CPTJYRY5A/p1648616316576289

Related external issues:
* https://github.com/googleapis/nodejs-bigquery/issues/799

The issue was fixed by just setting `projectId` in `BigQuery`'s client lib.

Now a new issue came up when try to insert rows:
```
Access Denied: BigQuery BigQuery: Streaming insert is not allowed in the free tier
```

I've also tried using the BigQuery UI and when I try to make an insert I get also this message
```
Billing has not been enabled for this project. Enable billing at https://console.cloud.google.com/billing. DML queries are not allowed in the free tier. Set up a billing account to remove this restriction.
```

Update:
In order to test the component it's necessary to enable billing for the project in google clound account

### CODE REVIEW

File: components/google_cloud/utils/base.mjs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    return JSON.parse(valueToParse);
+  } catch (e) {
+    throw new ConfigurationError("Make sure the custom expression contains a valid object");
+  }
+}
+
+export default {
+  emptyStrToUndefined,
+  parse,
+};
```
Comment (lcaresia) [inline components/google_cloud/utils/base.mjs:25]: Rename this file to `utils.mjs` and move to `common` folder.

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution.

Just some minor changes!

**Review by lcaresia — DISMISSED** [review]
LGTM!

**Comment by vunguyenhung** [comment]
Hello @jcortes, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[BigTable3070.pdf](https://github.com/PipedreamHQ/pipedream/files/8871215/BigTable3070.pdf)

**Comment by jcortes** [comment]
> Hello @jcortes, I have tested this PR and there're some test cases failed or needed improvement.
> 
> Please check the test report below for more information [BigTable3070.pdf](https://github.com/PipedreamHQ/pipedream/files/8871215/BigTable3070.pdf)

Hi @vunguyenhung can you please try to test again with latest change. Thanks in adance!

**Comment by vunguyenhung** [comment]
Hello @jcortes, the source can load more rows now on the first run
screencapture-pipedream-sources-dc-2EuRawb-2022-06-13-20_18_43

Though for new rows (that I have added via Pipedream Google Cloud - Insert BigQuery action), there is no new event emitted
Google Big Query
Screen Shot 2022-06-13 at 20 23 23

Pipedream Source events
screencapture-pipedream-sources-dc-2EuRawb-2022-06-13-20_24_07

Pipedream Source log
screencapture-pipedream-sources-dc-2EuRawb-logs-2022-06-13-20_24_19

Could you check again?

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Google_Cloud_3070.pdf](https://github.com/PipedreamHQ/pipedream/files/8892190/Google_Cloud_3070.pdf)

**Review by dannyroosevelt — APPROVED** [review]

**Pull request merged**